astropy/io/misc/hdf5.py

```diff
@@ -140,9 +145,15 @@ def read_table_hdf5(input, path=None):
     table = Table(np.array(input))
 
     # Read the meta-data from the file
-    if META_KEY in input.attrs:
-        header = meta.get_header_from_yaml(
-            h.decode('utf-8') for h in input.attrs[META_KEY])
+    old_version_meta = META_KEY in input.attrs
```

Comment:
Add a comment describing the reason for this old/new split and mentioning the astropy version where it was first included.